#! /bin/sh /usr/share/dpatch/dpatch-run
## 62_classes_crash.dpatch by Kenshi Muto <kmuto@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: TODO: forward upstream, find details

@DPATCH@
diff -urNad cups-1.2.3~/scheduler/classes.c cups-1.2.3/scheduler/classes.c
--- cups-1.2.3~/scheduler/classes.c	2006-02-22 22:43:17.000000000 +0000
+++ cups-1.2.3/scheduler/classes.c	2006-09-15 12:33:25.000000000 +0000
@@ -279,6 +279,11 @@
     if (i >= c->num_printers)
       i = 0;
 
+    if (c->num_printers == 0) {
+      cupsdLogMessage(CUPSD_LOG_ERROR, "Class \"%s\" hasn't any printers!", name);
+      return (NULL);
+    }
+
     if (c->printers[i]->accepting &&
         (c->printers[i]->state == IPP_PRINTER_IDLE ||
          ((c->printers[i]->type & CUPS_PRINTER_REMOTE) && !c->printers[i]->job)))
